### PR TITLE
Implement Authentication with google

### PR DESCRIPTION
**Pull request by HanifZafari**
Authentication:
- Added login page and Implemented Google Sign-In functionality
- Restricted authentication to Monash domain `@student.monash.edu` emails only
- Created a `ProtectedRoute` component for securing routes
- Added error handling and user redirection on the login page
- Implemented a sign-out dropdown menu
- Integrated team and role information into the global auth state

Documentation:
- Updated README with information on authentication, auth store usage, and protected routes

### CODE REVIEW

File: frontend/src/assets/RegisterPage.jpg
Comment (Rahul-Saxena-1) [inline frontend/src/assets/RegisterPage.jpg:1]: Can't see the image on the register screen atm

**Review by Rahul-Saxena-1 — APPROVED** [review]
Amazing work! Approved with comments

File: frontend/src/assets/RegisterPage.jpg
Comment (HanifZafari) [reply frontend/src/assets/RegisterPage.jpg:1]: I have used it on the login screen.

File: frontend/src/util/supabaseClient.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// supabaseClient.js
+import { createClient } from "@supabase/supabase-js";
+
+const supabaseUrl = "https://bcnxifsqkyzzpgshacts.supabase.co";
+const supabaseKey =
+	"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImJjbnhpZnNxa3l6enBnc2hhY3RzIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTU1NjM5NjQsImV4cCI6MjAzMTEzOTk2NH0.XT_aSkBLaaaX-66BqDF7Z1oP5qlgJLGNXSovibjatdU";
+
+export const supabase = createClient(supabaseUrl, supabaseKey);
```
[the remark this answers is not in the document]
Comment (HanifZafari) [reply frontend/src/util/supabaseClient.ts]: No, I setup this when prototyping and forgot to switch to environment variables. This has been fixed.

**Review by sairamishetty — APPROVED** [review]
Looking good!

**Review by sai-ramishetty — APPROVED** [review]

**Pull request merged**